orbit_determination/ekf.py

```diff
@@ -162,41 +174,39 @@ def measurement(
                 ]
             )
         )
+        # Iterated Update
+        for i in range(num_iter):
 
-        # Select a fraction of the measurements to use to speed up computations
-        z0 = z[0][: int(math.ceil(z[0].shape[0] * 0.05))]
-        z1 = z[1][: int(math.ceil(z[1].shape[0] * 0.05))]
-
-        h = self.h_est(z1, data_manager, x_p)
-        H = self.h_jac(z1, data_manager, x_p)
-
-        # Flatten the measurement vector
-        z0 = np.array(z0.reshape(-1))
-
-        # Let R take the dimensionality of the number of measurements
-        self.R = np.diag([1e-5] * z0.shape[0])
+            h = self.h_est(z1, data_manager, x_p)
+            H = self.h_jac(z1, data_manager, x_p)
+            S = H @ self.P_p @ H.T + self.R
 
-        S = H @ self.P_p @ H.T + self.R
+            # Check for ill-conditioned matrix and add regularization if necessary
+            if i == 0:
+                cond = np.linalg.cond(S)
+                if cond > self.cond_threshold:
+                    S += np.eye(S.shape[0]) * 1e-6
+                    print("Ill-conditioned matrix detected. Regularization applied.")
 
-        # Check for ill-conditioned matrix and add regularization if necessary
-        cond = np.linalg.cond(S)
-        if cond > self.cond_threshold:
-            S += np.eye(S.shape[0]) * 1e-6
-            print("Ill-conditioned matrix detected. Regularization applied.")
+            K = self.P_p @ H.T @ np.linalg.inv(S)
 
-        K = self.P_p @ H.T @ np.linalg.inv(S)
+            delta = K @ (z0 - h)
```

Comment:
Are you sure about this? right now it looks like we are computing the delta as K*(residual). 

I believe it should be K*(residual + H*delta)
See equation.9-84 of Maybeck 1979. Also see eq.22 of "Implementation of a Map Relative Localization System for Planetary Landing" (DOI: 10.2514/1.G006780)